### ISSUE TITLE
Need models for 3d output

### ISSUE THREAD

**Issue by softegg** [Spritetm/hadbadge2019_pcb#1]
I'm trying to design an enclosure in advance of Hackaday Superconference, and although I can export a the PCB as a 3d model,  a lot of the 3d models are missing. It is much easier to construct an accurate enclosure if the parts that need holes or space in the enclosure, or button / switch tops are available and accurate.

A lot of the 3d models for parts like switches and the BGA part don’t seem to be part of the standard KiCAD distribution, and Googling isn’t turning them up.
So far, I’m looking at:
The buttons: SW_PUSH_6mm_H9.5mm.wrl The notes give a Taobao link, I suppose I should look it up, but it would be nice to not have to model it from scratch.
The connector for the screen (FH26W-45S-0.3SHW(60)) seems to have no model even listed, and there is not even a part on the board for the screen, so it seems. Found the part number (ZYX Technology LQ032ALJV01) in the BOM notes, so I looked that up. Although the manufacturer has a link, all it does is let you request a model. 
The USB connector on J2: Connector_USB.3dshapes doesn’t contain USB_Micro-B_Amphenol, just other brands.
HDMI (!?) on J4 has no model. Seems to be Boom Precision HDMI-001.
The power switch doesn’t have a 3d model. I had to search to find what I think is a model number “Hroparts K3 12965”. I guess I have to build that one. Is it similar to some other part I might be able to find?
The BGA (not that important to model around, but nice for completeness): Lattice_caBGA-381_17.0x17.0mm_Layout20x20_P0.8mm_Ball0.4mm_Pad0.4mm_NSMD.wrl
The battery holders don’t have 3d models or model numbers, but they seem pretty generic, so maybe I can just use a standard one from something else and hope it fits.
The IR Receiver at U12 has no 3d model set. Seems to be RPM973-H11
Finally, speaker attached to J3, it would be good to know the part number

If some of these could be provided, it would make life a lot easier for me!

**Comment by Spritetm** [comment]
FYI, I didn't add any 3d models to the PCB design myself, as I didn't need that feature. (I'd even have to look up how to do 3d models in the first place...) I'm afraid that what is in the repo is what you get...

**Comment by softegg** [comment]
It just seemed like there were some parts that had models linked, implying that they exist, that aren't part of the standard distribution, and aren't included here.  Googling didn't turn them up. So I thought they might be living on your machine somehow, and perhaps you could share them?

**Comment by Spritetm** [comment]
If they are, I have no idea how those links got there, as I said I didn't add any models myself and my Kicad install is just a Debian apt-get install jobbie which I assume just pulls in the default libs.

**Comment by softegg** [comment]
Can I get measurements on the display at least? Or a link to a datasheet with such? (And maybe the visible area?) 
I've been googling, but I can't seem to find anything about LQ032ALJV01. I've found LQ35 and greater, but not this particular one.

**Comment by Spritetm** [comment]
Yeah, the display is some obscure Chinesium, but hey, it was cheap and looks pretty good. I can probably give you measurements when I'm back home, this evening. Any other things you want me to put a pair of calipers against?

**Comment by softegg** [comment]
Well, I've been substituting 3d models for other similar parts in absence of the actual parts and am pretty close to done save for the screen. I suspect that my HDMI port is probably too tall, so measuring that would be kind.
I'm currently using the model at https://grabcad.com/library/fci-hdmi-right-angle-header-1 in that place.
Also, I see in your notes that the manufacturer substituted a different button than the 4.3mm one you specified. Can you tell what size they actually used?

**Comment by Spritetm** [comment]
HDMI connector has a datasheet [here](https://lcsc.com/product-detail/Audio-Video-Connectors_Boom-Precision-Elec-HDMI-001_C138388.html). (Link is on there, not sure if I can link it directly.)

LCD: Does [this sketch](http://j0h.nl/XPcB/full) give enough info? Note that the LCD is affixed to the PCB with double-sided foam tape of unknown thickness; I don't have any final badges here so I can't measure that.

**Comment by softegg** [comment]
![hadbadge2019 front](https://user-images.githubusercontent.com/5379773/68766748-52e27580-05d4-11ea-90f8-6e41ffc4e735.png)
![hadbadge2019 back](https://user-images.githubusercontent.com/5379773/68766756-5675fc80-05d4-11ea-817e-62e1c82059f9.png)
![Hackaday Supercon 2019 Badge Enclosure Complete Front](https://user-images.githubusercontent.com/5379773/68766760-58d85680-05d4-11ea-95da-bcc53d4406dc.png)
Made some progress!
How tall did the stem on the buttons end up being?
They seem quite close together...
I'm a little worried about the ports... they will be quite recessed in the case and I worry about the enclosure interfering with the cable housing.
I still need to model the backs of the buttons, lightpipe for LEDs, cartridge port and I/O access, but it is starting to look like something.

**Comment by softegg** [comment]
Can I get the size of the stem on the buttons, please?
Also, is there a design for the cartridge PCB?

**Comment by sphereinabox** [comment]
The buttons are 9.5mm or so tall when not clicked. 

https://github.com/Spritetm/hadbadge2019_protoboard has the protoboard

**Comment by softegg** [comment]
Thanks! I'll update the models. I'm already 3d printing them at the 4.3mm size, so I'll have to start the new print before I leave for Supercon. Maybe the current ones can be drilled out or something

Sorry I missed the presentation on them this morning; I was up till 6am working on getting this print going, and am still trying to finish the rest of the parts. I'll be there by 4pm, I hope!

![Supercon 2019 Badge Enclosure Front](https://user-images.githubusercontent.com/5379773/68979982-00f84600-07b4-11ea-813f-5096916270b8.jpg)
![Supercon 2019 Badge Enclosure Most Parts 1st print](https://user-images.githubusercontent.com/5379773/68979984-00f84600-07b4-11ea-8586-26c70e9af820.jpg)
